vir/src/legacy/cfg/visitor.rs

```diff
@@ -113,6 +127,26 @@ impl CfgMethod {
         });
     }
 
+    /// Mutably visit each expression used in a statement or successor.
+    /// Note: sub-expressions of expressions will not be visited.
+    pub fn walk_expressions_mut<F: FnMut(&mut Expr)>(&mut self, mut walker: F) {
+        self.walk_statements_mut(|s| {
+            // Replace *s with a dummy statement
+            let stmt = std::mem::replace(s, Stmt::Inhale(true.into()));
+            *s = stmt.map_expr(|mut e| {
+                walker(&mut e);
+                e
+            });
```

Comment:
This would be more efficient if we had a `visit_expressions_mut` method that executes a `FnMut(&mut Expr)` closure on each expression in a statement.